crates/bevy_reflect/src/map.rs

```diff
@@ -4,23 +4,32 @@ use bevy_utils::HashMap;
 
 use crate::{serde::Serializable, Reflect, ReflectMut, ReflectRef};
 
-/// An ordered `ReflectValue->ReflectValue` mapping. `ReflectValue` `Keys` are assumed to return a
-/// non-`None` hash. Ideally the ordering is stable across runs, but this is not required.
-/// This corresponds to types like [`std::collections::HashMap`].
+/// An ordered `ReflectValue->ReflectValue` mapping.
+///
+///`ReflectValue` `Keys` are assumed to return a non-`None` hash. Ideally the
```

Comment:
What does "ideally" mean here? What happens if they're not?

Reply:
This was language I retained from the previous docs -- my guess is that the only visible effect is unstable serialization order between runs. Since `DynamicMap` has been an aHash-based `HashMap` under the hood since its introduction, it seems evident that stable ordering is not meant to be a guarantee of the API.

TL;DR I'm happy to remove that language if it's not relevant!

Reply:
This makes a lot of sense to me. If we can verify that assumption, I would be very happy to add this explanation to the docs; it's useful information!